Source/WebCore/platform/graphics/PlatformDisplay.h

```diff
@@ -105,7 +105,8 @@ class PlatformDisplay {
 #endif
 
 #if USE(ATSPI)
-    void setAccessibilityBusAddress(String&& address) { m_accessibilityBusAddress = WTFMove(address); }
+    void setAccessibilityBusProxyAddress(String&& address) { m_accessibilityBusProxyAddress = WTFMove(address); }
+    const String accessibilityBusProxyAddress() const;
```

Comment:
I don't think this is right, the platform display only cares about the actual address to use, the same way we override the env var, whether it's the actual one or the socket one is a detail the display doesn't need to know.

Reply:
Hm, then I think I need to move all the code that computes the accessibility bus address to someplace else. Will consider this.

Reply:
I don't think PlatformDisplay really has anything to do with the accessibility bus at all. We've just put the code to compute the accessibility bus address here because it's a convenient global location, and it seems as nice a place for it to go as anywhere else does, so I'm not inclined to try moving it.

The problem is that XDGDBusProxy overrides the address here with an address pointing to the proxy socket instead, but we still need the original unmodified address later on when launching the xdg-dbus-proxy process. So the problem to solve is: do not overwrite the original address. I thought separating the original address from the proxy address was a nice way to do it, but any other way would work just fine. My next proposal will be a new function `PlatformDisplay::originalAccessibilityBusAddress` that works the same as `PlatformDisplay::accessibilityBusAddress` except it ignores the value set with `PlatformDisplay::setAccessibilityBusAddress`. That should work. I wanted to call it `PlatformDisplay::hostAccessibilityBusAddress`, but that would be confusing when running in the web process, because they'll return the same result in the web process, which never sees the real host bus address, only the proxy address.

Reply:
Updated, does this look OK to you now? And if not, do you have an alternative suggestion?

Reply:
The reason why it was added to PlatformDisplay is because in X11, the bus address is set as an atom in the x11 display (AT_SPI_BUS), see PlatformDisplayX11::platformAccessibilityBusAddress().